src/library_sdl.js

```diff
@@ -2412,14 +2425,20 @@ var LibrarySDL = {
           SDL.fillWebAudioBufferFromHeap(ptr, sizeSamplesPerChannel, soundBuffer);
           // Workaround https://bugzilla.mozilla.org/show_bug.cgi?id=883675 by setting the buffer only after filling. The order is important here!
           source['buffer'] = soundBuffer;
-          
+
+          source['onended'] = removeSource;
+          sources.push(source);
+
           // Schedule the generated sample buffer to be played out at the correct time right after the previously scheduled
           // sample buffer has finished.
           var curtime = SDL.audioContext['currentTime'];
 #if ASSERTIONS
-          if (curtime > SDL.audio.nextPlayTime && SDL.audio.nextPlayTime != 0) {
-            console.log('warning: Audio callback had starved sending audio by ' + (curtime - SDL.audio.nextPlayTime) + ' seconds.');
+          var pauseTime = SDL.audio.pauseStartTime !== undefined ? curtime - SDL.audio.pauseStartTime : 0;
```

Comment:
I think we don't need to introduce new state variables `SDL.audio.pauseStartTime` at all, if below instead of the line `SDL.audio.pauseStartTime = SDL.audioContext['currentTime'];` we set `SDL.audio.nextPlayTime = 0;` when pausing? That would be more elegant since it would be more stateless.